### PR TITLE
doctests for running-external-programs.md, [ci skip]

### PR DESCRIPTION
**Pull request by fredrikekre**
(`make check-whitespace` and the doctests pass locally, remove `[ci skip]` from commit message if/when this is merged)

### CODE REVIEW

File: doc/src/manual/running-external-programs.md [hunk 1]
```diff
@@ -228,7 +234,7 @@ The result is that a single line is printed: "hello | sort". Inside of backticks
 a literal pipe character. How, then, does one construct a pipeline? Instead of using "|" inside
```
Comment (tkelman) [inline doc/src/manual/running-external-programs.md]: I guess this needs some rewording with the deprecation of having them unescaped

File: doc/src/manual/running-external-programs.md [hunk 2]
```diff
@@ -254,18 +260,18 @@ that shells cannot.
 
 Julia can run multiple commands in parallel:
 
-```julia
+```jldoctest
 julia> run(`echo hello` & `echo world`)
-world
 hello
+world
```
Comment (tkelman) [inline doc/src/manual/running-external-programs.md]: this should be a non-deterministic order, as described below

Comment (fredrikekre) [inline doc/src/manual/running-external-programs.md]: Oh, then the example might be better the "wrong order". Will fix

File: doc/src/manual/running-external-programs.md (hunk 1 above)
Comment (fredrikekre) [inline doc/src/manual/running-external-programs.md]: I updated, not sure I got it right though

File: doc/src/manual/running-external-programs.md [hunk 3]
```diff
@@ -215,20 +221,21 @@ easily and safely just examine its interpretation without doing any damage.
 
 ## Pipelines
 
-Shell metacharacters, such as `|`, `&`, and `>`, are not special inside of Julia's backticks:
-unlike in the shell, inside of Julia's backticks, a pipe is always just a pipe:
+Shell metacharacters, such as `|`, `&`, and `>`, needs to be quoted (or escaped) inside of Julia's backticks:
```
Comment (tkelman) [inline doc/src/manual/running-external-programs.md]: need to be

Comment (fredrikekre) [inline doc/src/manual/running-external-programs.md]: fixed

File: doc/src/manual/running-external-programs.md [hunk 4]
```diff
@@ -215,20 +221,21 @@ easily and safely just examine its interpretation without doing any damage.
 
 ## Pipelines
 
-Shell metacharacters, such as `|`, `&`, and `>`, are not special inside of Julia's backticks:
-unlike in the shell, inside of Julia's backticks, a pipe is always just a pipe:
+Shell metacharacters, such as `|`, `&`, and `>`, need to be quoted (or escaped) inside of Julia's backticks:
```
Comment (tkelman) [inline doc/src/manual/running-external-programs.md]: Looks good. @StefanKarpinski do we want to say anything here about why we're enforcing this?

**Pull request merged**